source/common/filesystem/inotify/watcher_impl.h

```diff
@@ -42,6 +43,7 @@ class WatcherImpl : public Watcher, Logger::Loggable<Logger::Id::file> {
   int inotify_fd_;
   Event::FileEventPtr inotify_event_;
   std::unordered_map<int, DirectoryWatch> callback_map_;
+  InstanceImplPosix file_system_;
```

Comment:
In case the filesystem has state in the future, I think we should be getting this through the `Api` interface and not have a concrete filesystem here? Same in similar cases elsewhere.

Reply:
We trust the incoming change follows the pattern you expected? Please unresolve if this isn't the case.